### PR TITLE
Get Graph API version-compatible

### PR DESCRIPTION
**Pull request by phildini**
- Allow a version parameter that's respected by the request
- Add a permissions fetch to the GraphAPI

Tested by fetching permissions and publishing event to facebook.

### CODE REVIEW

File: src/facebook.py [hunk 1]
```diff
@@ -203,6 +206,19 @@ def delete_object(self, id):
...
 
+    def get_permissions(self, id):
+        args = {
+            'access_token': self.access_token
+        }
+        path = "{}/permissions".format(id)
+
+        data = self.request(path)
+        perms = {}
+        for permission in data['data']:
```
Comment (tophevb) [inline src/facebook.py]: Is 'data' guaranteed to be a key here or should this use `data.get('data')`?

File: src/facebook.py [hunk 2]
```diff
@@ -203,6 +206,19 @@ def delete_object(self, id):
         """Deletes the object with the given ID from the graph."""
         return self.request(id, post_args={"method": "delete"})
 
+    def get_permissions(self, id):
```
Comment (tophevb) [inline src/facebook.py]: Please add docstring explaining the structure of returned dict `perms`.

File: src/facebook.py [hunk 3]
```diff
@@ -216,7 +232,9 @@ def request(self, path, args=None, post_args=None):
             else:
                 args["access_token"] = self.access_token
         post_data = None if post_args is None else urllib.urlencode(post_args)
-        file = urllib.urlopen("https://graph.facebook.com/" + path + "?" +
+
+        url = self.url if not self.version else "{}{}/".format(self.url, self.version)
```
Comment (tophevb) [inline src/facebook.py]: Consider using named parameters for string formatting. It improves readability, particularly when theres more than one parameter.  (minor detail)

File: src/facebook.py (hunk 1 above)
Comment (phildini) [inline src/facebook.py]: guaranteed.

File: src/facebook.py [hunk 4]
```diff
@@ -87,8 +87,11 @@ class GraphAPI(object):
     for the active user from the cookie saved by the SDK.
     """
 
-    def __init__(self, access_token=None):
+    def __init__(self, access_token=None, version=None):
         self.access_token = access_token
+        self.version = version
+        self.host = "graph.facebook.com/"
+        self.url = "https://{}".format(self.host)
```
Comment (cubes) [inline src/facebook.py]: Maybe call this `self.base_url`, and precompute the value here, instead of repeatedly computing this in the request method?

**Comment by tophevb** [comment]
Cubes makes a good point, I feel like having a `self.url` and `self.base_url` redundant.  I'd say just build one url in init and use that everywhere.

Thanks for adding the docstring!  Minor url cleanup and ship it!

**Comment by cubes** [comment]
:ship:

**Pull request merged**